fix: re-enable ptpl and allow for not using reservations

tiagolobocastro opened the pull request:
chore: allow for not using reservations

This gives back control to the api client.
To do this, set the reservation parameters to "0".

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

fix(nvme-resv): setup ptpl from the caller side

Previously this was done incorrectly only when props was None.
It's better to leave it for the caller to decide.

tiagolobocastro committed:
chore: allow for not using reservations

This gives back control to the api client.
To do this, set the reservation parameters to "0".

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

mtzaurus commented on io-engine/src/bdev/nexus/nexus_bdev.rs:
```diff
@@ -211,6 +211,12 @@ impl NexusNvmeParams {
         self.preempt_policy = preempt_policy;
     }
+    /// Check if reservations are disabled.
+    pub fn disabled(&self) -> bool {
```
Can we use here a name like 'reservations_disabled()' to denote that it deals with reservations ?
Also, as a nit, 'enabled()' looks a bit more natural to me than '.disabled()', but it just a suggestion here.

tiagolobocastro replied:
Ah I thought the type name here was NvmeReservations which is why I left out reservations but looks like it's only NvmeParams, so I'll rename to reservations_enabled().

mtzaurus commented:
Can we use here a name like 'reservations_disabled()' to denote that it deals with reservations ?
Also, as a nit, 'enabled()' looks a bit more natural to me than '.disabled()', but it just a suggestion here.

tiagolobocastro committed:
fix(nvme-resv): setup ptpl from the caller side

Previously this was done incorrectly only when props was None.
It's better to leave it for the caller to decide.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

chriswldenyer reviewed (approved):
Looks reasonable

tiagolobocastro commented:
bors merge

tiagolobocastro committed:
fix: improve nexus creation grpc status message

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge

tiagolobocastro committed:
fix(allowed-hosts): store lvol allowed-hosts within the lvol properties

When an lvol (replica) is shared through PoolImport it may get automatically shared.
In this case, make sure we set the allowed-hosts previously stored in the lvol
extended attributes.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
bors merge
